cpp/src/io/csv/csv_gpu.cu

```diff
@@ -453,28 +440,27 @@ struct decode_op {
   template <typename T,
             typename std::enable_if_t<std::is_integral<T>::value and !std::is_same<T, bool>::value>
               * = nullptr>
-  __host__ __device__ __forceinline__ bool operator()(const char *data,
-                                                      void *out_buffer,
+  __host__ __device__ __forceinline__ bool operator()(void *out_buffer,
                                                       size_t row,
-                                                      long start,
-                                                      long end,
+                                                      char const *begin,
+                                                      char const *end,
                                                       ParseOptions const &opts,
                                                       column_parse::flags flags)
   {
     auto &value{static_cast<T *>(out_buffer)[row]};
 
     // Check for user-specified true/false values first, where the output is
     // replaced with 1/0 respectively
-    const size_t field_len = end - start + 1;
-    if (serializedTrieContains(opts.trueValuesTrie, data + start, field_len)) {
+    const size_t field_len = end - begin + 1;
+    if (serializedTrieContains(opts.trueValuesTrie, begin, field_len)) {
       value = 1;
-    } else if (serializedTrieContains(opts.falseValuesTrie, data + start, field_len)) {
+    } else if (serializedTrieContains(opts.falseValuesTrie, begin, field_len)) {
       value = 0;
     } else {
       if (flags & column_parse::as_hexadecimal) {
-        value = decode_value<T, 16>(data, start, end, opts);
+        value = decode_value<T, 16>(begin, end, opts);
       } else {
-        value = decode_value<T>(data, start, end, opts);
+        value = decode_value<T>(begin, end, opts);
       }
     }
```

Comment:
I like the IILE that is done in line ~304 of cpp/src/io/json/json_gpu.cu (there are a few places that this change can be made :+1: